src/EnergyPlus/StandardRatings.cc

```diff
@@ -1410,55 +1382,43 @@ namespace StandardRatings {
-                if ((DesignHeatingRequirementMin - StandardDesignHeatingRequirement(StandardDHRNum)) >
-                    (StandardDesignHeatingRequirement(StandardDHRNum + 1) - DesignHeatingRequirementMin)) {
-                    DesignHeatingRequirementMin = StandardDesignHeatingRequirement(StandardDHRNum + 1);
+        for (StandardDHRNum = 0; StandardDHRNum < TotalNumOfStandardDHRs - 1; ++StandardDHRNum) {
+            if (StandardDesignHeatingRequirement[StandardDHRNum] <= DesignHeatingRequirementMin &&
+                StandardDesignHeatingRequirement[StandardDHRNum + 1] >= DesignHeatingRequirementMin) {
+                if ((DesignHeatingRequirementMin - StandardDesignHeatingRequirement[StandardDHRNum]) >
+                    (StandardDesignHeatingRequirement[StandardDHRNum + 1] - DesignHeatingRequirementMin)) {
+                    DesignHeatingRequirementMin = StandardDesignHeatingRequirement[StandardDHRNum + 1];
                 } else {
-                    DesignHeatingRequirementMin = StandardDesignHeatingRequirement(StandardDHRNum);
+                    DesignHeatingRequirementMin = StandardDesignHeatingRequirement[StandardDHRNum];
                 }
             }
         }
-        if (StandardDesignHeatingRequirement(1) >= DesignHeatingRequirementMin) {
-            DesignHeatingRequirement = StandardDesignHeatingRequirement(1);
-        } else if (StandardDesignHeatingRequirement(TotalNumOfStandardDHRs) <= DesignHeatingRequirementMin) {
-            DesignHeatingRequirement = StandardDesignHeatingRequirement(TotalNumOfStandardDHRs);
+        if (StandardDesignHeatingRequirement[0] >= DesignHeatingRequirementMin) {
+            DesignHeatingRequirement = StandardDesignHeatingRequirement[0];
+        } else if (StandardDesignHeatingRequirement[TotalNumOfStandardDHRs - 1] <= DesignHeatingRequirementMin) {
+            DesignHeatingRequirement = StandardDesignHeatingRequirement[TotalNumOfStandardDHRs - 1];
         } else {
             DesignHeatingRequirement = DesignHeatingRequirementMin;
         }
 
-        for (BinNum = 1; BinNum <= TotalNumOfTemperatureBins(RegionNum); ++BinNum) {
-
-            if (RegionNum == 1) {
-                FractionalBinHours = RegionOneFracBinHoursAtOutdoorBinTemp(BinNum);
-            } else if (RegionNum == 2) {
-                FractionalBinHours = RegionTwoFracBinHoursAtOutdoorBinTemp(BinNum);
-            } else if (RegionNum == 3) {
-                FractionalBinHours = RegionThreeFracBinHoursAtOutdoorBinTemp(BinNum);
-            } else if (RegionNum == 4) {
-                FractionalBinHours = RegionFourFracBinHoursAtOutdoorBinTemp(BinNum);
-            } else if (RegionNum == 5) {
-                FractionalBinHours = RegionFiveFracBinHoursAtOutdoorBinTemp(BinNum);
-            } else if (RegionNum == 6) {
-                FractionalBinHours = RegionSixFracBinHoursAtOutdoorBinTemp(BinNum);
-            }
```

Comment:
Collapsing all of those arrays into a single array allows us to eliminate these IF/ELSE ladders for lookups. 🎩  tip @amirroth for spotting that.